### PR TITLE
docs: describe required ACLs for all commands

### PR DESCRIPTION
**Pull request by tgross**
Fixes https://github.com/hashicorp/nomad/issues/7475

I think some kind of table that maps from ACLs to commands might be useful as well (under the new Operations header), but I wasn't able to make that legible. We can do that in future documentation work.

Website preview: https://nomad-rhxcqvgd6.vercel.app/docs

### CODE REVIEW

File: command/acl_policy_list.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ Usage: nomad acl policy list
 
   List is used to list available ACL policies.
 
+  This command requires a management ACL token to view all policies. A
+  non-management token can be used to list policies associated with that
+  token.
```
Comment (notnoop) [inline command/acl_policy_list.go:23]: What's the specific capability required here?  It'd be nice to list it like the rest.

**Review by notnoop — APPROVED** [review]
Awesome.  I did a sample check for some commands, and they were all looking good. Thanks!

File: command/acl_policy_list.go (hunk 1 above)
Comment (tgross) [reply command/acl_policy_list.go:23]: That seems weird at first glance, but there isn't one. The token gets access to any policy that's been applied to that token via the [`-policy`](https://nomad-edohxyewc.vercel.app/docs/commands/acl/token-update#policy) flag on the token's create/update. Maybe the wording here needs work?

Comment (tgross) [reply command/acl_policy_list.go:23]: (Or in other words, the token can read its own policy.)

Comment (notnoop) [reply command/acl_policy_list.go:23]: Oh I see - my english is failing - can we write something "A non-management token can query its own policies", even though the token doesn't quite do the querying.

Comment (tgross) [reply command/acl_policy_list.go]: Reworded that for clarity.

File: website/pages/docs/commands/acl/policy-delete.mdx [hunk 2]
```diff
@@ -18,6 +18,8 @@ nomad acl policy delete <policy_name>
 
 The `acl policy delete` command requires the policy name as an argument.
 
+This command requires a `management` ACL token.
```
Comment (drewbailey) [inline website/pages/docs/commands/acl/policy-delete.mdx:21]:
```suggestion
This command requires a management ACL token.
```

File: website/pages/docs/commands/acl/policy-apply.mdx [hunk 3]
```diff
@@ -19,6 +19,8 @@ nomad acl policy apply [options] <name> <path>
 The `acl policy apply` command requires two arguments, the policy name and path
 to file. The policy can be read from stdin by setting the path to "-".
 
+This command requires a `management` ACL token.
```
Comment (drewbailey) [inline website/pages/docs/commands/acl/policy-apply.mdx:22]:
```suggestion
This command requires a management ACL token.
```

**Review by drewbailey — APPROVED** [review]
LGTM!

File: website/pages/docs/commands/acl/policy-delete.mdx (hunk 2 above)
Comment (drewbailey) [reply website/pages/docs/commands/acl/policy-delete.mdx:21]: saw that these were the only two marked up

**Pull request merged**